### PR TITLE
[mcs] Replace NET_2_1 by MOBILE

### PR DESCRIPTION
**Pull request by ludovic-henry**
NET_2_1 is always defined whenever we define MOBILE, so they have the same effect. .NET 2.1 was Silverlight but it's been dead for a while now, and we do not even support it.

cc @akoeplinger

### CODE REVIEW

File: mcs/build/profiles/mobile.make [hunk 1]
```diff
@@ -16,7 +16,6 @@ DEFAULT_REFERENCES = -r:$(topdir)/class/lib/$(PROFILE)/mscorlib.dll
 PROFILE_MCS_FLAGS = \
 	-d:NET_1_1 \
 	-d:NET_2_0 \
-	-d:NET_2_1 \
```
Comment (akoeplinger) [inline mcs/build/profiles/mobile.make]: We should probably leave this in, just as a safeguard if these defines are used somewhere else.

**Comment by akoeplinger** [comment]
I like it 👍

File: mcs/build/profiles/mobile.make (hunk 1 above)
Comment (marek-safar) [inline mcs/build/profiles/mobile.make]: I am not sure, we can keep it now but I'd still like to remove it later as it'll become confusing what is used and what is dead

Comment (ludovic-henry) [inline mcs/build/profiles/mobile.make]: I have no strong opinion on removing NET_2_1 in this PR. But for the sake of having as little breaking change as possible, I will leave the removing of it for another PR.
In the same note, but for yet another PR, I am pretty sure we could get rid of NET_1_1, NET_2_0 and NET_3_5 too.

**Pull request merged**